instrumentation/google-http-client-1.19/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/googlehttpclient/GoogleHttpClientTests.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.googlehttpclient;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.api.client.http.HttpRequest;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.instrumentation.testing.InstrumentationTestRunner;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpClientTestOptions;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpClientTestOptionsBuilder;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpClientTestServer;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpClientTests;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpClientTypeAdapter;
+import io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.DynamicTest;
+
+// TODO: This replaces the abstract client test base class
+public class GoogleHttpClientTests {
+
+  private final HttpClientTests<HttpRequest> delegate;
+  private final HttpClientTypeAdapter<HttpRequest> adapter;
+  private final InstrumentationTestRunner testRunner;
+  private final HttpClientTestServer server;
+
+  private GoogleHttpClientTests(
+      HttpClientTests<HttpRequest> delegate,
+      HttpClientTypeAdapter<HttpRequest> adapter,
+      InstrumentationTestRunner testRunner,
+      HttpClientTestServer server) {
+    this.delegate = delegate;
+    this.adapter = adapter;
+    this.testRunner = testRunner;
+    this.server = server;
+  }
+
+  public static GoogleHttpClientTests create(
+      HttpClientTypeAdapter<HttpRequest> adapter,
+      InstrumentationTestRunner testRunner,
+      HttpClientTestServer server) {
+    HttpClientTestOptions options = buildOptions();
+    HttpClientTests<HttpRequest> clientTests =
+        new HttpClientTests<>(testRunner, server, options, adapter);
+    return new GoogleHttpClientTests(clientTests, adapter, testRunner, server);
+  }
+
+  List<DynamicTest> all() {
+    return Stream.concat(delegate.all(), Stream.of(errorTracesWhenExceptionIsNotThrown()))
+        .collect(Collectors.toList());
+  }
+
+  DynamicTest errorTracesWhenExceptionIsNotThrown() {
```

Comment:
Just an idea: perhaps we could leave the inheritance and abstract test classes on the "local" (instrumented library) level? This could be a normal `@Test` then.
My idea here is that a "local" inheritance hierarchy usually doesn't have that many moving parts (there's usually like 3 classes in total), and usually is not that hard to understand -- opposite to having an abstract class that is shared across the whole project, in which case there's lots of overridable options and the inheritance hierarchy gets quite abysmal. It's a compromise, basically.